realscale.py

```diff
@@ -101,10 +180,20 @@ def effect(self):
         
         # calculate scaling center
         center = self.calc_scale_center(p1_x, p1_y, p2_x, p2_y)
+        drawnscale = int(self.options.drawnscale)
 
         #calculate scaling factor
         target_length = self.getUnittouu(str(self.options.length) + self.options.unit)
-        factor = target_length / p_length
+        #inkex.debug("%s, %s" % (target_length, p_length))
+        factor = (target_length / p_length) / drawnscale
+        
+        # Add scale rule
+        if self.options.showscale == "true":
+            dist = int(self.options.unitlength)
+            # !! this isn't working right if drawnscale != 1 (1:1)
```

Comment:
Sorry, I didn't even get to test it yet... And I'm maybe a little bit too conservative in wanting to have a version that works according to the specs in the main branch at all times... :/ (i.e. if the non-working parts would be commented out, I'd merge... is that impolite?... This is the first github project I have where actually there's someone else contributing, I honestly don't know.)